interface-definitions/interfaces-tinc.xml.in

```diff
@@ -0,0 +1,476 @@
+              </valueHelp>
+              <constraint>
+                <regex>(off|internal|kernel)</regex>
+              </constraint>
+            </properties>
+            <defaultValue>internal</defaultValue>
+          </leafNode>
+          <leafNode name="graph-dump-file">
+            <properties>
+              <help>Dump network graph files, which can be read with graphviz</help>
+            </properties>
+          </leafNode>
+          <leafNode name="resolve-hostname">
+            <properties>
+              <help>Resolve IP addresses (real and VPN)</help>
+              <valueless/>
+            </properties>
+          </leafNode>
+          <leafNode name="iff-One-Queue">
+            <properties>
+              <help>Set IFF_ONE_QUEUE flag on TUN/TAP devices</help>
+              <valueless/>
+            </properties>
+          </leafNode>
+          <leafNode name="key-expire">
+            <properties>
+              <help>Key expiration time(seconds,default:3600)</help>
+            </properties>
+            <defaultValue>3600</defaultValue>
+          </leafNode>
+          <leafNode name="local-discovery">
+            <properties>
+              <help>Enable Local Discovery</help>
+              <valueless/>
+            </properties>
+          </leafNode>
+          <leafNode name="mac-expire">
+            <properties>
+              <help>MAC expiration time(seconds,default:600)</help>
+            </properties>
+            <defaultValue>600</defaultValue>
+          </leafNode>
+          <leafNode name="max-timeout">
+            <properties>
+              <help>Max Timeout(seconds,default:900)</help>
+            </properties>
+            <defaultValue>900</defaultValue>
+          </leafNode>
+          <leafNode name="ping-interval">
+            <properties>
+              <help>Ping Interval(seconds,default:60)</help>
+            </properties>
+            <defaultValue>60</defaultValue>
+          </leafNode>
+          <leafNode name="ping-timeout">
+            <properties>
+              <help>Ping Timeout(seconds,default:5)</help>
+            </properties>
+            <defaultValue>5</defaultValue>
+          </leafNode>
+          <leafNode name="priority-inheritance">
+            <properties>
+              <help>Enable Priority Inheritance</help>
+              <valueless/>
+            </properties>
+          </leafNode>
+          <leafNode name="private-keyfile">
+            <properties>
+              <help>Private Key File,This is the full path name of the RSA private key file that was generated by ‘tincd --generate-keys’. It must be a full path, not a relative directory(require)</help>
+            </properties>
+          </leafNode>
+          <leafNode name="hosts-dir">
+            <properties>
+              <help>The directory where the network interface stores host certificates(require)</help>
+            </properties>
+          </leafNode>
```

Comment:
> This is not what i ment. All hosts (remote nodes) needs to be defined with its own configuration files and encryption public keys. these are not defined anywhere as i can see. the ConnectTo option only specifies what primary endpoint should be tried to connect to. all connect statements also needs a corresponding host config

Reply:
There are more options required on a host, not only certificate. these needs to be added to the cli so the user do not need to manually craft configuration files for each host. see here: https://www.tinc-vpn.org/documentation/Host-configuration-variables.html#Host-configuration-variables

Reply:
These configuration files usually contain certificate information, so they are not well separated

Reply:
In addition to the configuration of the local node, the hosts records of other nodes usually come from the peer server, so they directly obtain their configuration certificate files. In this case, no secondary settings should be made

Reply:
i'm not following you.. if you have two vyos devices thatyou set up with tinc between each other. where is the host files created? in this case the user need to manually craft each file becase there is no way to sync configs between the vyos'es, and the host config is not saved anywhere on the local device by the vyos system. And this is no good. so no. i think the host config file needs to be possible to create in the configuration system of the vyos device. and the cli can have the option to support both cli generated configs and syncing them from other devices.

Reply:
At present, I have set up to automatically generate the hosts configuration file of the local host. Some parameters of the local node already exist in the configuration node (such as `subnets`,`local_address`,`port`), file synchronization on other devices. I consider using `scp` command, which can be directly copied through the operation mode command